# Copyright (c) 2020  Carnegie Mellon University
#
# Permission is hereby granted, free of charge, to any person obtaining a copy
# of this software and associated documentation files (the "Software"), to deal
# in the Software without restriction, including without limitation the rights
# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
# copies of the Software, and to permit persons to whom the Software is
# furnished to do so, subject to the following conditions:
#
# The above copyright notice and this permission notice shall be included in all
# copies or substantial portions of the Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
# SOFTWARE.

main_menu:
  type: list
  title: MAIN_MENU
  items:
#    -
#      menu: destination_menu
#    -
#      menu: exploration_menu
    -
      menu: speech_speed_menu
    -
      menu: speech_voice_menu
    -
      menu: volume_menu
    -
      menu: max_velocity_menu
    -
      menu: developer_menu

exploration_menu:
  type: item
  title: LAUNCH_EXPLORATION_MODE
  format: EXPLORATION_MODE_IS_LAUNCHED

speech_speed_menu:
  type: adjust
  title: SPEECH_SPEED
  usage: MENU_ADJUST_USAGE
  name: speech_speed
  min: 0
  max: 100
  step: 5
  default: 50

speech_voice_menu:
  type: adjust
  title: SPEECH_VOICE
  usage: MENU_ADJUST_USAGE
  name: speech_voice
  min: 0
  max: 1
  step: 1
  default: 0
  values: ["FEMALE", "MALE"]

volume_menu:
  type: adjust
  actions:
    -
      type: syscommand
      command: "pactl set-sink-mute @DEFAULT_SINK@ off; pactl set-sink-volume @DEFAULT_SINK@ %d%%"
  title: SYSTEM_VOLUME
  usage: MENU_ADJUST_USAGE
  name: system_volume
  min: 0
  max: 100
  step: 5
  default: 50

max_velocity_menu:
  type: adjust
  actions:
    -
      type: publish_topic
      msg_type: std_msgs.msg.Float32
      topic: /cabot/user_speed
  title: MAXIMUM_VELOCITY
  usage: MENU_ADJUST_USAGE
  name: max_velocity
  min: 0.0
  max: 1.0
  step: 0.05
  default: 0.5
  format: "{:.2f}"


